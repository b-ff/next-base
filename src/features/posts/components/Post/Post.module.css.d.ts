declare const styles: {
  readonly "lastModified": string;
};
export = styles;

