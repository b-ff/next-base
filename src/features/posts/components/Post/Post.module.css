.lastModified {
  font-size: var(--font_size_small);
  color: var(--color_font_neutral);
}
