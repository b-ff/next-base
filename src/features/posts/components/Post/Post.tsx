import { FC } from "react";
import { Flex } from "@/components/Flex";
import { Post as PostModel } from "@/schema";

import styles from "./Post.module.css";

export type PostProps = {
  post: PostModel;
};

/**
 * Post representation
 */
export const Post: FC<PostProps> = (props) => {
  const { post } = props;

  const date = new Intl.DateTimeFormat().format(
    Math.max(
      new Date(post.createdAt)?.getTime(),
      new Date(post.updatedAt)?.getTime()
    )
  );

  return (
    <article>
      <Flex gap={4}>
        <h2>{post.title}</h2>
        <small className={styles.lastModified}>{date}</small>
        <p>{post.body}</p>
      </Flex>
    </article>
  );
};
