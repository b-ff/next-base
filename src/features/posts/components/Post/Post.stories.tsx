import type { Meta, StoryObj } from "@storybook/react";

import { Post } from "./Post";
import { Post as PostModel } from "@/schema";

const meta: Meta<typeof Post> = {
  component: Post,
  tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof Post>;

const fakePost: PostModel = {
  id: 1,
  userId: "that-exact-user-id",
  title: "For whom the bell tolls?",
  body: "For those who refuse to take action and cross the boundaries. For those who complaining instead of doing things. For them we shall fire the torch and spread the light.",
  createdAt: new Date(),
  updatedAt: new Date(),
};

export const Primary: Story = {
  args: {
    post: fakePost,
  },
};
