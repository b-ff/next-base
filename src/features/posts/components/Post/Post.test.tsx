import { Post } from "./Post";
import { render } from "@testing-library/react";

const mockPost = {
  id: 1,
  userId: "that-exact-user-id",
  title: "For whom the bell tolls?",
  body: "For those who refuse to take action and cross the boundaries. For those who complaining instead of doing things. For them we shall fire the torch and spread the light.",
  createdAt: new Date("2023-06-20T14:00:00Z"),
  updatedAt: new Date("2023-06-20T14:00:00Z"),
};

describe("Component Post", () => {
  it("Should render as expected", async () => {
    const { container } = render(<Post post={mockPost} />);
    expect(container).toMatchSnapshot();
  });
});
