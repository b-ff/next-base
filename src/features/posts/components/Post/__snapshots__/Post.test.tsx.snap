// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Post Should render as expected 1`] = `
<div>
  <article>
    <section
      style="display: flex; flex-direction: column; gap: var(--spacing_4x);"
    >
      <div>
        <h1>
          For whom the bell tolls?
        </h1>
      </div>
      <div>
        <small
          class="lastModified"
        >
          20 June 2023 at 14:00
        </small>
      </div>
      <div>
        <p>
          For those who refuse to take action and cross the boundaries. For those who complaining instead of doing things. For them we shall fire the torch and spread the light.
        </p>
      </div>
    </section>
  </article>
</div>
`;
